### ISSUE TITLE
When building nuget packages with a nuspec using $configuration$ in it is a null or empty string resuling in failures.

### ISSUE THREAD

**Issue by AraHaan** [dotnet/sdk#15407]
I made an nuspec file which then I had to provide a specific copy for debug / release just because of the fact that it seems $configuration$ is not passed to the nuspec somehow and then package making fails to pack in files due to it not being set.

I also tried within the csproj on the files individually with set to ``bin\$(Configuration)\$(Platform)\$(TargetFramework)\[the output file name]`` on the packing file path and the path to put it in on the nupkg file itself however then on the generated nuspec it would generate this:
``[path to the project folder]\bin\\AnyCPU\net5.0\``

Resulting in:

```C:\Program Files\dotnet\sdk\5.0.102\Sdks\NuGet.Build.Tasks.Pack\buildCrossTargeting\NuGet.Build.Tasks.Pack.targets(207,5): error : Could not find a part of the path '[path to the project folder]\bin\\AnyCPU\net5.0\'```

The issue also occurs on the broken .NET 5 Preview SDK on the latest Visual Studio 2019 Preview as well so then I had to revert to 5.0.102 temporarily to fix it.

Currently the only workaround is to hard code the Platform and configuration values on my own provided nuspec until this gets patched.

Basically what I am trying to do is something similar to a metapackage except:
- does not reference everything else as a package, instead adds them as files.
- everything that compiles ProjectReferences an msbuild task to test the task that would get packaged into it as well too.

For more information see Elskom/Sdk#22 (and related code is at https://github.com/Elskom/Sdk/blob/master/Elskom.Sdk/Elskom.Sdk.dbg.nuspec and https://github.com/Elskom/Sdk/blob/master/Elskom.Sdk/Elskom.Sdk.rel.nuspec)

On my local device I unified them in an attempt that then suffers from this issue.

[Elskom.Sdk.nuspec.zip](https://github.com/dotnet/sdk/files/5839654/Elskom.Sdk.nuspec.zip)

**Comment by AraHaan** [comment]
cc @dsplaisted

**Comment by dsplaisted** [comment]
@zkat @nkolev92 When running the pack target, is there a way to pass the configuration through to a nuspec file that is specified with the `NuspecFile` property in MSBuild?  See https://github.com/Elskom/Sdk/blob/f4fed059a2c68b489829b4d9a5567a9e3af4d1f0/Elskom.Sdk/Elskom.Sdk.csproj#L24-L35

**Comment by nkolev92** [comment]
Yep, `NuspecProperties`

https://docs.microsoft.com/en-us/nuget/reference/msbuild-targets#packing-using-a-nuspec has an example.

**Comment by dsplaisted** [comment]
@AraHaan Then what I think you want to add is this:

```xml
<NuspecProperties>Configuration=$(Configuration)</NuspecProperties>
```
